Retire torch.distributed.pipeline

kwen2501 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #127354

Actually retiring module after deprecation warning for a while.
The new supported module is: torch.distributed.pipelining. 
Please migrate.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @ezyang @gchanan @albanD

kwen2501 committed:
Retire torch.distributed.pipeline

[ghstack-poisoned]

wconstab reviewed (approved):
oops i submitted a parallel PR to remove old pipeline.rst. ill cancel mine.

albanD commented on docs/source/conf.py:
```diff
@@ -606,29 +606,22 @@
     # torch.distributed.optim.utils
     "as_functional_optim",
     "register_functional_optim",
-    # torch.distributed.pipeline.sync.checkpoint
```
Please delete the corresponding entries! Not just the comment saying which module they come from!

kwen2501 replied:
hahahaha, sorry my bad, I thought those are real entries, but just commented out :)

kwen2501 committed:
Update on "Retire torch.distributed.pipeline"


Actually retiring module after deprecation warning for a while.
The new supported module is: torch.distributed.pipelining. 
Please migrate.

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k ezyang gchanan albanD

[ghstack-poisoned]

kwen2501 committed:
Update on "Retire torch.distributed.pipeline"


Actually retiring module after deprecation warning for a while.
The new supported module is: torch.distributed.pipelining. 
Please migrate.

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k ezyang gchanan albanD

[ghstack-poisoned]

kwen2501 commented:
@pytorchbot merge -f "Minor change to fix conflict. All tests have passed previously"

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your change but the doc build failure looks legit https://hud.pytorch.org/pytorch/pytorch/commit/b9c058c203ee38032594f898f27cd8404f113a63' -c ignoredsignal

kwen2501 committed:
Update on "Retire torch.distributed.pipeline"


Actually retiring module after deprecation warning for a while.
The new supported module is: torch.distributed.pipelining. 
Please migrate.

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k ezyang gchanan albanD

[ghstack-poisoned]

kwen2501 commented:
@pytorchbot merge -f "Minor doc fix; CI should pass now"

kwen2501 committed:
Update on "Retire torch.distributed.pipeline"


Actually retiring module after deprecation warning for a while.
The new supported module is: torch.distributed.pipelining. 
Please migrate.

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k ezyang gchanan albanD

[ghstack-poisoned]

kwen2501 commented:
@pytorchbot merge -f "rebase"

kwen2501 commented:
@pytorchbot merge -f "this is a rebase to resolve merge conflict in doc"
